lib/ret_web/channels/hub_channel.ex

```diff
@@ -343,35 +365,57 @@ defmodule RetWeb.HubChannel do
     socket.assigns |> Map.take([:presence, :profile, :context])
   end
 
-  defp join_with_hub(%Hub{entry_mode: :deny}, _socket, _endpoint, _auth_token) do
+  defp join_with_hub(%Hub{entry_mode: :deny}, _account, _socket, _params) do
     {:error, %{message: "Hub no longer accessible", reason: "closed"}}
   end
 
-  defp join_with_hub(%Hub{} = hub, socket, push_subscription_endpoint, auth_token) do
+  defp join_with_hub(%Hub{} = hub, %Account{}, _socket, %{
+         hub_requires_oauth: true,
+         account_has_provider_for_hub: false
+       }),
+       do: require_oauth(hub)
+
+  defp join_with_hub(%Hub{}, %Account{}, _socket, %{
+         hub_requires_oauth: true,
+         account_has_provider_for_hub: true,
```

Comment:
i'd put the denies first so they are the most aggressively covered, to reduce risk of regressions